perses/rjmc/topology_proposal.py

```diff
@@ -467,7 +473,9 @@ def _get_mol_atom_map(molA,
 
             scaffold_A_maps = AtomMapper._get_all_maps(molA, scaffoldA,
                                      atom_expr=oechem.OEExprOpts_AtomicNumber,
-                                     bond_expr=0)
+                                     bond_expr=0,
+                                     allow_ring_breaking=allow_ring_breaking,
```

Comment:
I don't think we would ever want allow_ring_breaking to be True for the scaffold mapping

Reply:
or to use matching_criterion here

Reply:
@dominicrufa  could you comment on why we had allow_ring_breaking=True originally?

Reply:
I think you would still want it for the general maps, but not for the scaffold mapping. The scaffold mapping should just be mappings of the 'main' part of the molecule, not for any side chains, so I don't think we would want to compare scaffolds with different ring structures? I don't think it should really change the function overall though